core/src/test/java/org/elasticsearch/common/network/NetworkUtilsTests.java

```diff
@@ -74,4 +78,24 @@ public void testFilter() throws Exception {
         assertArrayEquals(new InetAddress[] { InetAddress.getByName("127.0.0.1") }, NetworkUtils.filterIPV4(addresses));
         assertArrayEquals(new InetAddress[] { InetAddress.getByName("::1") }, NetworkUtils.filterIPV6(addresses));
     }
+
+    /**
+     * Test that selecting by name is possible and properly matches the addresses on all interfaces and virtual
+     * interfaces.
+     *
+     * Note that to avoid that this test fails when interfaces are down or they do not have addresses assigned to them,
+     * they are ignored.
+     */
+    public void testAddressInterfaceLookup() throws Exception {
+        for (NetworkInterface netIf : NetworkUtils.getInterfaces()) {
+            if (!netIf.isUp() || Collections.list(netIf.getInetAddresses()).isEmpty()) {
+                continue;
```

Comment:
wouldn't this change the semantics though in that if lets say the first one is down all the other potential test-able interfaces would be ignored?